ladybug_geometry/geometry2d/polygon.py

```diff
@@ -313,10 +313,62 @@ def is_valid(self):
         """
         return not self.area == 0
 
+    def is_equivalent(self, other, tolerance):
+        """ Boolean noting equivalence (based on point tolerance) between this polygon and
+        another polygon. The order of the polygon vertices do not have to start from the
+        same vertex for equivalence to be true, but must be in the same counterclockwise
+        or clockwise order.
+
+        Args:
+            other: Polygon2D for comparison.
+            tolerance: float representing point tolerance.
+
+        Returns:
+            True if equivalent else False
+        """
+
+        # Check number of points
+        if len(self.vertices) != len(other.vertices):
+            return False
+
+        vertices = list(self.vertices)
+
+        # Check order
+        if not vertices[0].is_equivalent(other.vertices[0], tolerance):
+            self_idx = None
+            other_pt = other.vertices[0]
+            for i, pt in enumerate(self.vertices):
+                if pt.is_equivalent(other_pt, tolerance):
+                    self_idx = i
+                    break
+
+            if self_idx is None:
+                return False
+
+            # Re-order polygon vertices to match othe
+            vertices = vertices[self_idx:] + vertices[:self_idx]
+
+        is_equivalent = True
+        for pt, other_pt in zip(vertices[1:], other.vertices[1:]):
+            is_equivalent = is_equivalent and pt.is_equivalent(other_pt, tolerance)
+        return is_equivalent
+
     def to_array(self):
         """Get a list of lists whenre each sub-list represents a Point2D vetex."""
         return tuple(pt.to_array() for pt in self.vertices)
 
+    @classmethod
+    def from_array(cls, point_array):
+        """ Consumes nested list of nested list of points,
+        and returns Polygon2D.
+
+        Args:
+            point_array: nested list of point lists
+        Returns:
+            Polygon2D
```

Comment:
Could you apply all of the comments that I had about the `LineSegment2D` classmethod docstring to this one?